clients/include/cblas_interface.hpp

```diff
@@ -1119,9 +1119,40 @@ inline void cblas_ger(rocblas_int m,
 }
 
 // syr
+
+// cblas_syr doesn't have complex support so implementation below for float/double complex
 template <typename T>
 void cblas_syr(
-    rocblas_fill uplo, rocblas_int n, T alpha, T* x, rocblas_int incx, T* A, rocblas_int lda);
+    rocblas_fill uplo, rocblas_int n, T alpha, T* xa, rocblas_int incx, T* A, rocblas_int lda)
+{
+    if(n <= 0)
+        return;
+
+    T* x = (incx < 0) ? xa - ptrdiff_t(incx) * (n - 1) : xa;
+
+    if(uplo == rocblas_fill_upper)
+    {
+        for(int j = 0; j < n; ++j)
+        {
+            T tmp = alpha * x[j * incx];
+            for(int i = 0; i <= j; ++i)
+            {
+                A[i + j * lda] = A[i + j * lda] + x[i * incx] * tmp;
```

Comment:
The docs [here](http://www.netlib.org/lapack/explore-html/d2/d15/group__complex_s_yauxiliary_gab7fbaf7c4477110907f682b250b24abe.html#gab7fbaf7c4477110907f682b250b24abe) have the operation as `A := alpha*x*x**H + A`. Is `x**H` not the conjugate transpose of `x`? In their code there it doesn't seem as they're conjugating it either which makes me wonder if I'm interpreting the docs incorrectly.

Reply:
I think you are interpreting the documents correctly but I hope just like ours they have bugs.   I am following the code.   Both cuda and magma docs say a transpose not **H ,  I better dig into the magma code to ensure this.   As I am writing the reference and function I better be sure.
The complex conjugates are in her, but of course that matrix is Hermitian too, this is a complex non-hermitian symmetric matrix.

Reply:
Also matches blis/flame as reference but this looks like a straight copy of lapack on which my cpu code was based.   See newer commit.  I think we can call this resolved agreed?

Reply:
Yup, that's fine with me. I'd maybe suggest stressing in the rocblas-functions.h documentation that it is still a transpose and **not** a conjugate-transpose in the complex case, but other than that this is good with me.

Reply:
Well it says A[i] := A[i] + alpha*x[i]*x[i]**T   ,  usually we say if an alternative **H is done for complex.   Should I say " for all data types including complex" or does that confuse?

Reply:
I guess it's probably ok as is if you think. If anyone wants to be sure they can always look at the implementation.

Reply:
I'll leave it as it correctly says what we do.